File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GoogleMap/Features/MapLocationPropertyPage.vb
Imports Gizmox.WebGUI.Forms.Google

Namespace CompanionKit.Controls.GoogleMap.Features

    Public Class MapLocationPropertyPage

        Public Sub New()
            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()
        End Sub

        ''' <summary>
        ''' Handles Click event for Button
        ''' </summary>
        Private Sub mobjSetCoordinatesButton_Click(sender As Object, e As EventArgs) Handles mobjSetCoordinatesButton.Click
            ' Set local variable for validation
            Dim isValid As Boolean = True
            ' Set local variables for coordinates
            Dim longitude As Double = CDbl(Me.mobjLongitudeNUD.Value)
            Dim latitude As Double = CDbl(Me.mobjLatitudeNUD.Value)

            ' If all values entered are valid
            If isValid Then
                ' Set map location for GoogleMap control
                Me.mobjGoogleMap.MapLocation = New Gizmox.WebGUI.Forms.Google.GoogleMapLocation(latitude, longitude)
            End If
        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GoogleMap/Features/MapLocationPropertyPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.GoogleMap.Features
{
    public partial class MapLocationPropertyPage : UserControl
    {
        public MapLocationPropertyPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles Click event for Button
        /// </summary>
        private void mobjSetCoordinatesButton_Click(object sender, EventArgs e)
        {
            // Set local variable for validation
            bool isValid = true;
            // Set local variables for coordinates
            double longitude = (double) this.mobjLongitudeNUD.Value;
            double latitude = (double) this.mobjLatitudeNUD.Value ;

            // If all values entered are valid
            if (isValid)
            {
                // Set map location for GoogleMap control
                this.mobjGoogleMap.MapLocation = new Gizmox.WebGUI.Forms.Google.GoogleMapLocation(latitude, longitude );
            }            
        }
    }
}